source/common/upstream/default_local_address_selector.cc

```diff
@@ -0,0 +1,33 @@
+#include "source/common/upstream/default_local_address_selector.h"
+
+#include <string>
+
+namespace Envoy {
+namespace Upstream {
+
+DefaultUpstreamLocalAddressSelector::DefaultUpstreamLocalAddressSelector(
+    std::vector<::Envoy::Upstream::UpstreamLocalAddress> upstream_local_addresses)
+    : upstream_local_addresses_(std::move(upstream_local_addresses)) {
+  ASSERT(upstream_local_addresses_.size() > 0);
```

Comment:
Was this previously a thrown exception? If so, let's keep the same behavior as before.

Reply:
Previously we were creating an `UpstreamLocalAddress` with `nullptr` for `address_` on the data plane if there was no `BindConfig`. I changed it to do it on the control plane [here](https://github.com/envoyproxy/envoy/pull/27705/files#diff-70f33a41a52fb3e38be32857512e0de967bf016ee43479ca1530aa86c9843790R276-R282)  in `upstream_impl.cc`

Thus `upstream_local_addresses_` is never expected to be empty.